pkg/operator/target_config_reconciler.go

```diff
@@ -223,6 +221,28 @@ func (c TargetConfigReconciler) sync() error {
 		specAnnotations["rolebindings/leader-election"] = resourceVersion
 	}
 
+	if roleBindings, _, err := c.manageRole(kueue, "assets/kueue-operator/role-prometheus.yaml"); err != nil {
+		klog.Error("unable to create role prometheus")
+		return err
+	} else {
+		resourceVersion := "0"
```

Comment:
what happens if it stays this version?

Reply:
I reviewed the manageRole() function and the underlying API call. I don't think we need the nil check after the resourceVersion, nor the `resourceVersion := 0` declaration at all.

Reply:
I did a refactoring to address this everywhere. https://github.com/openshift/kueue-operator/pull/37